File: src/main/java/cn/lnkdoc/sdk/uia/instance/bjtoon/util/sign/SignUtil.java
package cn.lnkdoc.sdk.uia.instance.bjtoon.util.sign;

import cn.lnkdoc.sdk.uia.instance.bjtoon.util.encryption.MD5Coder;
import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Map;


/**
 * AbstractUiaClient
 *
 * @author shizongwang
 * @version v 1.0 2017-12-28
 */
@SuppressWarnings(value = {"unused"})
public class SignUtil {

    /**
     * getter
     * 
     * @param params sign map
     * @return sign (String)
     */
    public static String getSignContent(Map<String, String> params) {
        StringBuilder content = new StringBuilder();
        List<String> keys = new ArrayList<>(params.keySet());
        Collections.sort(keys);
        int index = 0;
        for (String key : keys) {
            String value = params.get(key);
            //参数不为空的按照key=value用&拼接
            if (StringUtils.isNoneBlank(key, value)) {
                content.append((index == 0 ? "" : "&")).append(key).append("=").append(value);
                index++;
            }
        }
        return content.toString();

    }

    /**
     * getter
     * 
     * @param content content
     * @param signType signType
     * @return sign
     */
    public static String getSign(String content,String signType){

        if(StringUtils.equalsIgnoreCase("md5",signType)){
            return MD5Coder.MD5(content);
        }
        return null;

    }



}

File: awesome-uia-sdk-support/awesome-uia-bjtoon-sdk/src/main/kotlin/cn/lnkdoc/sdk/uia/instance/bjtoon/util/sign/SignUtil.kt
package cn.lnkdoc.sdk.uia.instance.bjtoon.util.sign

import cn.lnkdoc.sdk.uia.instance.bjtoon.util.encryption.MD5Coder
import org.apache.commons.lang3.StringUtils
import java.util.*


/**
 * AbstractUiaClient
 *
 * @author shizongwang
 * @version v 1.0 2017-12-28
 */
@Suppress("unused")
object SignUtil {
    /**
     * getter
     *
     * @param params sign map
     * @return sign (String)
     */
    @JvmStatic
    fun getSignContent(params: Map<String, String?>): String {
        val content = StringBuilder()
        val keys: List<String> = ArrayList(params.keys)
        //Collections.sort(keys)
        keys.sorted()
        var index = 0
        for (key in keys) {
            val value = params[key]
            //参数不为空的按照key=value用&拼接
            if (StringUtils.isNoneBlank(key, value)) {
                content.append(if (index == 0) "" else "&").append(key).append("=").append(value)
                index++
            }
        }
        return content.toString()
    }

    /**
     * getter
     *
     * @param content content
     * @param signType signType
     * @return sign
     */
    @JvmStatic
    fun getSign(content: String, signType: String?): String? {
        return if (StringUtils.equalsIgnoreCase("md5", signType)) {
            MD5Coder.MD5(content)
        } else null
    }
}

